src/Microsoft.AspNet.OData.Shared/DefaultContainerBuilder.cs

```diff
@@ -13,7 +13,29 @@ namespace Microsoft.AspNet.OData
     /// </summary>
     public class DefaultContainerBuilder : IContainerBuilder
     {
-        private readonly IServiceCollection services = new ServiceCollection();
+        private IServiceCollection services;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultContainerBuilder"/> class.
+        /// </summary>
+        public DefaultContainerBuilder()
+            : this(new ServiceCollection())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultContainerBuilder"/> class.
+        /// </summary>
+        /// <param name="services">IServiceCollection collection in which to add services.</param>
+        internal DefaultContainerBuilder(IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw Error.ArgumentNull("services");
+            }
```

Comment:
Or we could just create a new ServiceCollection()?

Reply:
DI 2.0 doesn't use IServiceCollection anymore though, right? Shouldn't it just be using a ServiceCollection?

Reply:
@Mike - This appears to be an unused refactor; AspNetCore supports DI natively and at one point I needed a ContainerBuilder to be wrapped around an existing service collection but it's not being used in either this code or the latest code. I'll remote it.

@robertmclaws - This version is still compiling against DI 1.1.1 but the current version (on my box) compiles against 2.0 without issue using IServiceCollection.